src/http/middleware.ts

```diff
@@ -89,3 +89,34 @@ export abstract class Middleware {
    */
   abstract run(): any | void;
 }
+
+export type middlewareToRun = {
+  before_request?: any[];
+  after_request?: any[];
+};
+export function runMiddleware(obj: middlewareToRun) {
```

Comment:
Wonder if it's possible to have a clean solution where this is named `middleware`? (Your approach might be able to do this compared to mine). @crookse might also need your input on this on how this might conflict with existing `middleware` methods/types etc and hopefully not cause any confusion

What about if we move server level middleware into decorators as well (outside of this feature maybe)? (think you mentioned it @crookse), that way it could reduce any conflicts or confusion with the `middleware` naming. And if possible, the `runMiddleware` method could check if the decorator is over a class or method

Reply:
I agree the naming should be revised. I would prefer to do it after #185. I already started messing with the branch and it will improve our typings a lot! I'll keep a look onto this.